util/parser.go

```diff
@@ -73,6 +76,23 @@ func FormatTimeToString(inputTime time.Time, format string) string {
 	return inputTime.Format(format)
 }
 
-func DateTimeToRecordFormat(inputTime time.Time) *string {
-	return StringValueOrNil(FormatTimeToString(inputTime, RecordDateTimeFormat))
+func DateTimeToRFC3339(inputTime interface{}) string {
+	var timeField time.Time
+	switch v := inputTime.(type) {
+
+	case *time.Time:
+		if reflect.ValueOf(inputTime).IsNil() {
+			return ""
+		}
+		timeField = *v
+
+	case time.Time:
+		timeField = v
+
+	default:
+		l.Log.Errorf("Unexpected data type: expected *time.Time or time.Time, got %s", v)
+		return ""
+	}
```

Comment:
I would like to hear the opinion of the rest of the team, but I personally would not give `DateTimeToRFC3339` the responsibility of having to parse the input. I think it would be better to extract this part into another function.

Reply:
I guess I'm confused - all this function is doing is transforming an interface from many types down to a time.Time that we can format. What would we pull out?

Reply:
I was thinking, that something like this you mean when you wrote in comment above "You should be able to make util.DateTimeToRFC3339() accept an interface as a param instead of a time.Time struct, and do the right thing based on the type. Just do a nil check and return nil if the time is nil."

Reply:
I forgot to add the `DateTimeToRFC3339` function in the code sample :man_facepalming: . My point was that, in my opinion, the `DateTimeToRFC3339`'s responsibility should only be "I take a time, and I convert its format to RFC 3339", not "I take something, try to see if its a valid `time.Time`, and if it is, convert its format to RFC 3339". But never mind, I was just asking what you two thought about it :+1: